### DISCUSSION TITLE
cannot import name 'FramerType' from 'pymodbus'

### DISCUSSION THREAD

**Discussion by sterlingStu** [pymodbus-dev/pymodbus#2192] [Help] [answered]
I'm using a fresh install of pymodbus on windows 11 (this is my first project on a new laptop). I've followed the docs and installed with "pip install pymodbus[serial]" and copied the "Simple synchronous client" code into a new document, but when I run the code I get the exception:
`Traceback (most recent call last):
  File "c:\Software\modbus-master\esp32-connector.py", line 17, in <module>
    from pymodbus import (
ImportError: cannot import name 'FramerType' from 'pymodbus' (C:\Users\me\AppData\Local\Programs\Python\Python311\Lib\site-packages\pymodbus\__init__.py)`
 I've tried this with both python 3.12.3 (latest) and 3.11.9 with no luck. Thanks in advance for any advice.

**Comment by janiversen** [comment, marked as answer]
You do not tell which version of pymodbus you are using ?

It looks a lot like you have installed an older version of pymodbus but have copied the example from latest ! you need the example version to match the pymodbus version.

**Reply by sterlingStu** [reply]
Thanks for getting back to me, I really appreciate it!

I have the following versions:
```
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pymodbus
>>> pymodbus.__version__
'3.6.8'
```
pip 24.0

Which I think are the latest. 

On Windows 11 Pro version 23H2

And I've copied the [Simple Synchronous Client](https://pymodbus.readthedocs.io/en/dev/source/examples.html#simple-synchronous-client) code into a new script.

**Reply by sterlingStu** [reply]
After changing all instances of "FramerType" to "Framer" I'm making headway

**Reply by janiversen** [reply]
But is only going to work until you update to v3.7.0

**Reply by sterlingStu** [reply]
Yep you're right, messing with the code is a bad route. I've pulled the github repository and I'm running through the examples now. It looks like the latest release from pip is at 3.6.8, but the example is ready and set up for 3.7. 

Thanks for the repository and your help. Beautiful library. Really appreciate all your efforts.

**Reply by janiversen** [reply]
dev is prepared for 3.7 !!! if you want to test 3.6.8, checkout master.

**Reply by janiversen** [reply]
I think the examples are identically between 3.6.8 and dev (current status of 3.7.0).

**Reply by sterlingStu** [reply]
Yup, I missed the option on pymodbus.readthedocs.io to change the version so did exactly what you said and copied the dev. example when using the release code. Apologies for the rookie mistake.
